src/server/service/debug.rs

```diff
@@ -311,4 +319,80 @@ impl debugpb_grpc::Debug for Service {
+        sink: UnarySink<RegionConsistentCheckResponse>,
+    ) {
+        let region_id = req.get_region_id();
+        let debugger = self.debugger.clone();
+        let router1 = self.raft_router.clone();
+        let router2 = self.raft_router.clone();
+
+        let consistency_check = future::result(debugger.get_store_id())
+            .and_then(move |store_id| region_detail(router2, region_id, store_id))
+            .and_then(|detail| consistency_check(router1, detail));
+        let f = self.pool
+            .spawn(consistency_check)
+            .map(|_| RegionConsistentCheckResponse::new());
+        self.handle_response(ctx, sink, f, "check_region_consistency");
+    }
+}
+
+fn region_detail<T: RaftStoreRouter>(
+    raft_router: T,
+    region_id: u64,
+    store_id: u64,
+) -> impl Future<Item = RegionDetailResponse, Error = Error> {
+    let mut header = RaftRequestHeader::new();
+    header.set_region_id(region_id);
+    header.mut_peer().set_store_id(store_id);
+    let mut status_request = StatusRequest::new();
+    status_request.set_cmd_type(StatusCmdType::RegionDetail);
+    let mut raft_cmd = RaftCmdRequest::new();
+    raft_cmd.set_header(header);
+    raft_cmd.set_status_request(status_request);
+
+    let (tx, rx) = oneshot::channel();
+    let cb = Callback::Read(box |resp| tx.send(resp).unwrap());
+    future::result(raft_router.send_command(raft_cmd, cb))
+        .map_err(|e| Error::Other(box e))
+        .and_then(move |_| {
+            rx.map_err(|e| Error::Other(box e)).and_then(move |mut r| {
+                if r.response.get_header().has_error() {
+                    let e = r.response.get_header().get_error();
+                    warn!("Debug::consistent-check got error: {:?}", e);
+                    let msg = print_to_string(e);
+                    return Err(Error::Other(msg.into()));
+                }
+                let detail = r.response.take_status_response().take_region_detail();
+                debug!("Debug::consistent-check got region detail: {:?}", detail);
+                let leader_store_id = detail.get_leader().get_store_id();
+                if leader_store_id != store_id {
+                    let msg = format!("Leader is on store {}", leader_store_id);
+                    return Err(Error::Other(msg.into()));
+                }
+                Ok(detail)
+            })
+        })
+}
+
+fn consistency_check<T>(raft_router: T, mut detail: RegionDetailResponse) -> Result<(), Error>
+where
+    T: RaftStoreRouter,
+{
+    let mut header = RaftRequestHeader::new();
+    header.set_region_id(detail.get_region().get_id());
+    header.set_peer(detail.take_leader());
+    let mut admin_request = AdminRequest::new();
+    admin_request.set_cmd_type(AdminCmdType::ComputeHash);
+    let mut raft_cmd = RaftCmdRequest::new();
+    raft_cmd.set_header(header);
+    raft_cmd.set_admin_request(admin_request);
+
+    raft_router
+        .send_command(raft_cmd, Callback::None)
```

Comment:
You may want to set a callback here to make sure the command doesn't get dropped.